### PR TITLE
[IAST] Add updated source names.

### PR DESCRIPTION
**Pull request by NachoEchevarria**
## Summary of changes

The source names have changed and we need to update the vulnerability json to the latest version, described in https://github.com/DataDog/experimental/blob/484e1cb819c2df594dda69742fa6aa3fb8f42f45/teams/asm/vulnerability_schema/vulnerability_schema.json#L17-L27

### CODE REVIEW

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (3988)](https://github.com/DataDog/dd-trace-dotnet/tree/1dd6db842f34863891a2745dc29cac2760cacc0c)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/930b0a8c605beffd9939dae6f6312cb69bc938c9)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (3988) - mean (3,009ms)  : 2883, 3135
     .   : milestone, 3009,
    master - mean (3,027ms)  : 2897, 3156
     .   : milestone, 3027,

    section CallTarget+Inlining+NGEN
    This PR (3988) - mean (3,656ms)  : 3573, 3739
     .   : milestone, 3656,
    master - mean (3,650ms)  : 3561, 3740
     .   : milestone, 3650,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (3988) - mean (3,129ms)  : 2955, 3304
     .   : milestone, 3129,
    master - mean (3,168ms)  : 3027, 3309
     .   : milestone, 3168,

    section CallTarget+Inlining+NGEN
    This PR (3988) - mean (3,523ms)  : 3418, 3628
     .   : milestone, 3523,
    master - mean (3,528ms)  : 3436, 3620
     .   : milestone, 3528,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (3988) - mean (3,127ms)  : 2958, 3297
     .   : milestone, 3127,
    master - mean (3,150ms)  : 3020, 3280
     .   : milestone, 3150,

    section CallTarget+Inlining+NGEN
    This PR (3988) - mean (3,477ms)  : 3400, 3555
     .   : milestone, 3477,
    master - mean (3,484ms)  : 3372, 3596
     .   : milestone, 3484,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (3988) - mean (188ms)  : 185, 190
     .   : milestone, 188,
    master - mean (187ms)  : 184, 191
     .   : milestone, 187,

    section CallTarget+Inlining+NGEN
    This PR (3988) - mean (921ms)  : 878, 965
     .   : milestone, 921,
    master - mean (912ms)  : 882, 943
     .   : milestone, 912,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (3988) - mean (366ms)  : 361, 371
     .   : milestone, 366,
    master - mean (365ms)  : 361, 370
     .   : milestone, 365,

    section CallTarget+Inlining+NGEN
    This PR (3988) - mean (1,051ms)  : 1029, 1073
     .   : milestone, 1051,
    master - mean (1,051ms)  : 1024, 1078
     .   : milestone, 1051,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (3988) - mean (355ms)  : 349, 362
     .   : milestone, 355,
    master - mean (354ms)  : 350, 358
     .   : milestone, 354,

    section CallTarget+Inlining+NGEN
    This PR (3988) - mean (1,000ms)  : 976, 1024
     .   : milestone, 1000,
    master - mean (1,005ms)  : 976, 1034
     .   : milestone, 1005,

```

**Comment by andrewlock** [comment]
## Throughput/Crank Report:zap:

Throughput results for AspNetCoreSimpleController comparing the following branches/commits:
- [This PR (3988)](https://github.com/DataDog/dd-trace-dotnet/tree/1dd6db842f34863891a2745dc29cac2760cacc0c)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/930b0a8c605beffd9939dae6f6312cb69bc938c9)
- [benchmarks/2.9.0](https://github.com/DataDog/dd-trace-dotnet/tree/0d56b1d9f25673f5257b1b7df97a3754d2df63f4)
- [benchmarks/2.27.0](https://github.com/DataDog/dd-trace-dotnet/tree/d9da4dfad8955dc19736a0e586478bfede45ce2a)

Cases where throughput results for the PR are worse than latest master (5% drop or greater), results are shown in **red**.

Note that these results are based on a _single_ point-in-time result for each branch. For full results, see [one](https://ddstaging.datadoghq.com/dashboard/fnz-afi-c2i/apm-net-crank) of the [many](https://ddstaging.datadoghq.com/dashboard/c5g-i7d-kcy/ciapp-apm-net-throughput-tests), [many](https://ddstaging.datadoghq.com/dashboard/uxh-m8j-qhi/ciapp-apm-net-throughput-tests-kevin) dashboards!

```mermaid
gantt
    title Throughput Linux x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (3988) (11.117M)   : 0, 11116709
    master (11.317M)   : 0, 11317104
    benchmarks/2.27.0 (11.348M)   : 0, 11347693
    benchmarks/2.9.0 (11.230M)   : 0, 11230274

    section Automatic
    This PR (3988) (7.886M)   : 0, 7885587
    master (7.859M)   : 0, 7858666
    benchmarks/2.27.0 (8.058M)   : 0, 8057813
    benchmarks/2.9.0 (8.195M)   : 0, 8194841

    section Trace stats
    master (7.928M)   : 0, 7928084
    benchmarks/2.27.0 (8.035M)   : 0, 8034872

    section Manual
    This PR (3988) (9.952M)   : 0, 9952250
    master (9.941M)   : 0, 9941294
    benchmarks/2.27.0 (10.027M)   : 0, 10026779

    section Manual + Automatic
    This PR (3988) (7.639M)   : 0, 7639254
    master (7.622M)   : 0, 7622364
    benchmarks/2.27.0 (7.745M)   : 0, 7745298

    section Version Conflict
    master (6.848M)   : 0, 6847771
    benchmarks/2.27.0 (6.906M)   : 0, 6905611

```
```mermaid
gantt
    title Throughput Linux arm64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (3988) (9.613M)   : 0, 9612651
    master (9.560M)   : 0, 9559622
    benchmarks/2.27.0 (9.606M)   : 0, 9606435
    benchmarks/2.9.0 (9.336M)   : 0, 9335873

    section Automatic
    This PR (3988) (6.774M)   : 0, 6774036
    master (6.867M)   : 0, 6866550
    benchmarks/2.27.0 (6.781M)   : 0, 6780786

    section Trace stats
    master (6.758M)   : 0, 6758387
    benchmarks/2.27.0 (6.872M)   : 0, 6871663

    section Manual
    This PR (3988) (8.524M)   : 0, 8524297
    master (8.476M)   : 0, 8476148
    benchmarks/2.27.0 (8.589M)   : 0, 8589048

    section Manual + Automatic
    This PR (3988) (6.471M)   : 0, 6470970
    master (6.620M)   : 0, 6620307
    benchmarks/2.27.0 (6.573M)   : 0, 6572742

    section Version Conflict
    master (5.953M)   : 0, 5953175
    benchmarks/2.27.0 (5.959M)   : 0, 5958642

```
```mermaid
gantt
    title Throughput Windows x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (3988) (9.740M)   : 0, 9739793
    master (9.988M)   : 0, 9987679
    benchmarks/2.27.0 (9.506M)   : 0, 9505940
    benchmarks/2.9.0 (9.650M)   : 0, 9649624

    section Automatic
    This PR (3988) (7.182M)   : 0, 7181750
    master (7.170M)   : 0, 7169523
    benchmarks/2.27.0 (7.004M)   : 0, 7004132
    benchmarks/2.9.0 (7.007M)   : 0, 7007478

    section Trace stats
    master (7.009M)   : 0, 7008587
    benchmarks/2.27.0 (6.847M)   : 0, 6846737

    section Manual
    This PR (3988) (8.544M)   : 0, 8544136
    master (8.491M)   : 0, 8491159
    benchmarks/2.27.0 (8.501M)   : 0, 8500521

    section Manual + Automatic
    This PR (3988) (6.790M)   : 0, 6789538
    master (6.777M)   : 0, 6777138
    benchmarks/2.27.0 (6.779M)   : 0, 6778511

    section Version Conflict
    master (6.054M)   : 0, 6054196
    benchmarks/2.27.0 (6.097M)   : 0, 6097497

```
```mermaid
gantt
    title Throughput Linux x64 (ASM) (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (3988) (7.527M)   : 0, 7527375
    master (7.477M)   : 0, 7476820
    benchmarks/2.27.0 (5.826M)   : 0, 5825827
    benchmarks/2.9.0 (7.826M)   : 0, 7826446

    section No attack
    This PR (3988) (2.167M)   : 0, 2167437
    master (2.206M)   : 0, 2206454
    benchmarks/2.27.0 (2.177M)   : 0, 2177280
    benchmarks/2.9.0 (3.305M)   : 0, 3304617

    section Attack
    This PR (3988) (1.848M)   : 0, 1847727
    master (1.776M)   : 0, 1775976
    benchmarks/2.27.0 (1.748M)   : 0, 1748479
    benchmarks/2.9.0 (2.546M)   : 0, 2546316

    section Blocking
    This PR (3988) (3.736M)   : 0, 3735807
    master (3.739M)   : 0, 3738671
    benchmarks/2.27.0 (3.548M)   : 0, 3548069

```

File: tracer/src/Datadog.Trace/IAST/SourceType.cs [hunk 1]
```diff
@@ -24,14 +24,17 @@ internal static string GetString(byte value)
...
             SourceTypeName.RequestHeader => "http.request.header",
             SourceTypeName.RequestHeaderName => "http.request.header.name",
-            SourceTypeName.RequestQueryString => "http.request.querystring",
+            SourceTypeName.RequestPath => "http.request.path",
+            SourceTypeName.RequestBody => "http.request.body",
+            SourceTypeName.RequestQuery => "http.request.query",
+            SourceTypeName.RoutedParameterValue => "http.request.path.parameter",
+            SourceTypeName.MatrixParameter => "http.request.matrix.parameter",
+            SourceTypeName.CookieName => "http.cookie.name",
+            SourceTypeName.CookieValue => "http.cookie.value",
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/IAST/SourceType.cs:37]: Doesn't look like these are used anywhere - do they need implementing (and testing)?

**Review by andrewlock — APPROVED** [review]

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #3988 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/647b93378e56c68cbbdd0f5e2765f49f26a96c4d) will **not change** line coverage
:heavy_check_mark: Merging #3988 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/647b93378e56c68cbbdd0f5e2765f49f26a96c4d) will **not change** branch coverage
:no_entry: Merging #3988 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/647b93378e56c68cbbdd0f5e2765f49f26a96c4d) will will **increase** complexity by `3`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/647b93378e56c68cbbdd0f5e2765f49f26a96c4d) | #3988       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `23487` / `33742` | `23516` / `33745` |          |
| Lines %   | `70%`      | `70%`      |  `0%` :heavy_check_mark:  |
| Branches  | `14281` / `22063` | `14301` / `22066` |          |
| Branches %| `65%`      | `65%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `23776`      | `23779`        |  `3`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=133608&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/133608/artifacts?artifactName=Code%20Coverage%20Report_133608&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3988](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=133651&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/133651/artifacts?artifactName=Code%20Coverage%20Report_133651&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/647b93378e56c68cbbdd0f5e2765f49f26a96c4d) | #3988       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `70%`      | `70%`       |  `0%` :heavy_check_mark:  |
| Branches %| `65%`      | `65%`       |  `0%` :heavy_check_mark:  |
| Complexity| `23776`      | `23779`       |  `3` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.DataStreamsMonitoring.Transport.DataStreamsApi](https://github.com/DataDog/dd-trace-dotnet/tree/e1f276e0f3796d91b513975fb04aadb9be2ac41f/tracer/src/Datadog.Trace/DataStreamsMonitoring/Transport/DataStreamsApi.cs) | `-17%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Iast.SourceType](https://github.com/DataDog/dd-trace-dotnet/tree/e1f276e0f3796d91b513975fb04aadb9be2ac41f/tracer/src/Datadog.Trace/IAST/SourceType.cs) | `-9%` :no_entry: | `-6%` :no_entry:   | `3` :no_entry: |
| [Datadog.Trace.Util.AsyncUtil](https://github.com/DataDog/dd-trace-dotnet/tree/e1f276e0f3796d91b513975fb04aadb9be2ac41f/tracer/src/Datadog.Trace/Util/AsyncUtil.cs) | `-9%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Configuration.CIGitMetadataTagsProvider](https://github.com/DataDog/dd-trace-dotnet/tree/e1f276e0f3796d91b513975fb04aadb9be2ac41f/tracer/src/Datadog.Trace/Ci/Configuration/CIGitMetadataTagsProvider.cs) | `0%` :heavy_check_mark: | `-30%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.GitInfo](https://github.com/DataDog/dd-trace-dotnet/tree/e1f276e0f3796d91b513975fb04aadb9be2ac41f/tracer/src/Datadog.Trace/Ci/GitInfo.cs) | `12%` :heavy_check_mark: | `7%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=133608&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/133608/artifacts?artifactName=Code%20Coverage%20Report_133608&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3988](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=133651&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/133651/artifacts?artifactName=Code%20Coverage%20Report_133651&api-version=6.0&%24format=zip)

**Review by pierotibou — APPROVED** [review]

**Review by robertpi — APPROVED** [review]

**Pull request merged**